Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -400,10 +347,139 @@ public void JumpMovement( Vector3 inputVec )
 		}
 	}
 
-	protected void ChangeState( ActorStates toState )
+	public void StartGlide()
+	{
+		if ( !_isOnGround )
+		{
+			ChangeState( ActorStates.Gliding );
+
+			rigidbody.useGravity = false;
+
+			if ( actor.animator )
+			{
+				actor.animator.SetBool( "isJumping", true );
+			}
+		}
+	}
+
+	public void EndGlide()
+	{
+		rigidbody.useGravity = true;
+
+		if ( actor.animator )
+		{
+			actor.animator.SetBool( "isJumping", false );
+		}
+	}
+
+	public void GlideMovement( Vector3 inputVec )
+	{
+		_moveVec = inputVec * _glideHorizontalSpeed;
+		_moveVec.y = -_glideDescentRate;
+
+		rigidbody.velocity = _moveVec;
+	}
+
+	public void RegisterStateMethod( ActorStates state, ActorStateMethod method )
+	{
+		_stateMethodMap.Add( state, method );
+	}
+
+	void ConstrainBumper()
+	{
+		_bumperTransform.position = transform.position;
+		_bumperRigidbody.velocity = rigidbody.velocity;
+	}
+
+	void FollowBumper()
+	{
+		if ( !IsInState( ActorStates.Climbing ) )
+		{
+			Vector3 constrainedPos = transform.position;
+			constrainedPos.x = _bumperTransform.position.x;
+			constrainedPos.z = _bumperTransform.position.z;
```

Comment:
you could flip this around and do constrainedPos = bumpertransform.position; constrainedPos.y = transform.position.y; transform.position = constrainedpos; Saves you one assignment. The current way might be more readable though.